### ISSUE TITLE
Unable to complete FaceTime

### ISSUE THREAD

**Issue by Supposedly-Sam** [airmessage/airmessage-server-next#2]
AirMessage is able to create a facetime but as soon as the other party accepts I get the error `AppleScript error -1719: System Events got an error: Can’t get button 2 of last group of list 1 of list 1 of scroll area 2 of window 1 of process "FaceTime". Invalid index.`
I am running a OnePlus 9 Pro running Android 11 2021-07-01
My sever is a Mac Mini late 2012 (MacMini6, 1) running MacOS 12.3 using OCLP.

**Comment by tagavari** [comment]
After AirMessage starts the FaceTime call, is the _Share link_ button in the sidebar clickable?

**Comment by andre4ik3** [comment]
I also have this issue. It never starts the call. I think it is because macOS *requires* a camera to even try to start a call.

**Comment by tagavari** [comment]
Yes, unfortunately a camera is required to use AirMessage's FaceTime integration. FaceTime links can't be generated from a FaceTime Audio call, so you would have to keep a camera plugged in.

**Comment by Supposedly-Sam** [comment]
Alright, I tried it again today and it brought up the facetime web interface. I could see the copy link button, but it said waiting to be let in. So now it looks like I'm having a problem being let in.

**Comment by tagavari** [comment]
Progress! AirMessage Server should automatically let you in when it detects you try to join.

Do you see your user show up on the sidebar when you join with your phone? It should show _your name (maybe)_ with the accept and reject buttons

**Comment by Supposedly-Sam** [comment]
I tried it again and everything worked. I wasn't patient enough.

Regarding my original issue: I believe it was caused by my mac's display going to sleep.

**Comment by Supposedly-Sam** [comment]
I tried it again and everything worked. I wasn't patient enough.

Regarding my original issue: I believe it was caused by my mac's display going to sleep.

**Comment by andre4ik3** [comment]
> Yes, unfortunately a camera is required to use AirMessage's FaceTime integration. FaceTime links can't be generated from a FaceTime Audio call, so you would have to keep a camera plugged in.

Is there any way around it? E.g. a virtual camera? I ordered a physical one just in case

Also I feel like this should be included in the docs

**Comment by tagavari** [comment]
I haven't done any testing myself, but from what I've researched FaceTime doesn't work with virtual cameras.

I definitely agree that the requirements should be documented better. I'm thinking of adding a dedicated page on the website, as well as a help button next to the option in the app's settings.

**Comment by Supposedly-Sam** [comment]
I use the app [Epoc Cam](https://www.elgato.com/en/epoccam) on my mac mini and facetime works

**Comment by Supposedly-Sam** [comment]
I just attempted a FaceTime today and I faced the same error. However, the error was easily fixed by closing and reopening FaceTime on my mac server

**Comment by larsdennert** [comment]
I get this too. The issue is after clicking get link, there is a variable delay, depending on the Mac or it's state, until the pop up menu appears. The script attempts the next click for a button that doesn't yet exist. There needs to be a check for it's existence first as the script only blocks during the click and has no way to block for a new menu that appears.
